integration-libs/cdc/core/profile/facade/cdc-user-register.service.ts

```diff
@@ -0,0 +1,106 @@
+import { Injectable } from '@angular/core';
+import { Store } from '@ngrx/store';
+import {
+  AuthService,
+  Command,
+  CommandService,
+  UserActions,
+  WindowRef
+} from '@spartacus/core';
+import { User } from '@spartacus/user/account/root';
+import {
+  Title, UserRegisterFacade, UserSignUp
+} from '@spartacus/user/profile/root';
+import {
+  UserProfileConnector, UserProfileService
+} from 'feature-libs/user/profile/core';
+import { CdcJsService } from 'integration-libs/cdc/root';
+import { Observable } from 'rxjs';
+import { tap } from 'rxjs/operators';
+
+@Injectable()
+export class CDCUserRegisterService implements UserRegisterFacade {
```

Comment:
As we discussed in the call, instead of overwriting UserRegisterFacade, let's introduce an intermediate layer of the `register-component.service`, similar to `login-form-component.service`. Thanks to that we don't need to implement the guest register method which doesn't make sense in the context of CDC. See the diagram below:
<img width="953" alt="image" src="https://user-images.githubusercontent.com/4001059/171006132-deedea85-1837-4d80-990d-e60204094683.png">

Reply:
agreed, will be creating a new `register-component.service.ts` file